### ISSUE TITLE
Renaming a variant doesn't delete the related options.value

### ISSUE THREAD

**Issue by chemicalkosek** [medusajs/admin#650]
When renaming a variant in the admin, the variant option doesn't get removed from option.values

Steps to reproduce:
1. Add option "Size" to product
2. Add two variants to product:
    - Title: "S", Size: 'S'
    - Title: "L", Size: 'L'
3. Rename L Variant to XL (size and title)
4. See it being renamed in the admin
5. The option value 'L' still exists in `product.options[0].values` which makes it appear on the storefront when displaying available options, despite that no such variant exists anymore.

**Comment by olivermrbl** [comment]
I can't seem to reproduce this issue running with `"@medusajs/medusa": "^1.3.3"`. Have performed the exact steps. Can I get you to share the version of Medusa, you are running with?

**Comment by chemicalkosek** [comment]
Hmm I can't reproduce it now either with new products and I tested it beforehand couple of times.

But I think I now understand what happened. I'm still using an old development database which had some products that were added before this fix: https://github.com/medusajs/medusa/issues/1464
Hence there were still some orphaned options on products and renaming one variant made other option reveal on the storefront. Because normalny they're hidden with this: 
https://github.com/medusajs/nextjs-starter-medusa/blob/dd1602584278f61b553cc8ae0ca6d82ba00a8a14/src/modules/products/components/option-select/index.tsx#L19

At least that's what I think happened. I'm sorry, I tripped on this. Will reopen if I reproduce this correctly.
